### ISSUE TITLE
Support for pipelines using Registry / via tuple

### ISSUE THREAD

**Issue by pbrooko** [dashbitco/broadway_kafka#57]
Hello!

I'm busy playing with Broadway for a use-case with many pipeline's that are created & destroyed dynamically. The changes introduced in dashbitco/broadway#239 seem perfect for this by allowing pipeline naming using a via tuple / Registry.

The `broadway_kafka` producer still assumes pipeline's are named using an atom, and tries to use `Module.concat/2` to build it's own child process names.

Is there a sane way to get the Kafka producer to also support naming its children using the pipeline's `process_name/2` function?

**Comment by josevalim** [comment]
Yes, if we depend on master we should be able to rely on the process_name API for the same result. A PR would be welcome!

**Comment by pbrooko** [comment]
Awesome, will see what I can put together..
